cocos/2d/CCSprite.cpp

```diff
@@ -269,9 +269,16 @@ bool Sprite::initWithTexture(Texture2D *texture, const Rect& rect, bool rotated)
         _recursiveDirty = false;
         setDirty(false);
 
-        _opacityModifyRGB = true;
-
-        _blendFunc = BlendFunc::ALPHA_PREMULTIPLIED;
+        if (!texture || !texture->hasPremultipliedAlpha())
+        {
+            _opacityModifyRGB = false;
+            _blendFunc = BlendFunc::ALPHA_NON_PREMULTIPLIED;
+        }
+        else
+        {
+            _opacityModifyRGB = true;
+            _blendFunc = BlendFunc::ALPHA_PREMULTIPLIED;
+        }
```

Comment:
`updateBlendFunc()` has the same logic.

Reply:
Updated both this PR and the PR for V4 to remove that redundant code.